core/open.py

```diff
@@ -0,0 +1,118 @@
+from dbt.config import RuntimeConfig
+
+import os
+import json
+import subprocess
+
+
+COMPILED_DIR = "compiled"
+RUN_DIR = "run"
+MANIFEST_FILE = "manifest.json"
+DEFAULT_OPEN_COMMAND = "open"
+
+
+class OpenTask:
+    def __init__(self, args):
+        self.args = args
+        self.config = RuntimeConfig.from_args(args)
+        self.model_path = self.config.source_paths[0]
+        self.target_path = self.config.target_path
+        self.manifest = self._get_manifest()
+        self.open_command = self._get_open_command()
+
+    def _get_open_command(self):
+        open_command = os.environ.get(
+            "DBT_HELPER_OPEN_COMMAND", os.environ.get("EDITOR", DEFAULT_OPEN_COMMAND)
+        )
+        return open_command
+
+    def _get_manifest(self):
+        """
+        This subcommand uses the manifest file, whereas other subcommands import
+        the manifest object from dbt (which requires the project to be parsed).
+        Using the manifest file is significantly faster, so is preferred in this
+        case.
+        """
+        manifest_path = os.path.join(self.target_path, MANIFEST_FILE)
+        try:
+            with open(manifest_path) as f:
+                manifest = json.load(f)
+            return manifest
+        except FileNotFoundError as err:
+            raise Exception(
+                "Could not find {} file. You may need to run dbt compile first".format(
+                    MANIFEST_FILE
+                )
+            )
+
+    def _get_model_file_names(self):
+        """ Return a dictionary of the form:
+        {
+            'my_model_1': {
+                'source': file_path,
+                'compiled': file_path,
+                'run': file_path
+            },
+            'my_model_2': {..}
+        }
+        """
+
+        file_names = {}
+        for node in self.manifest["nodes"].values():
```

Comment:
This is still a jank for-loop... idk if there's a better thing to be doing here

Reply:
I don't either! Let's leave it.